src/main/scala/pirate.internal/ParseTraversal.scala

```diff
@@ -130,33 +130,34 @@ object ParseTraversal {
       None
   }
 
-  def parseWord(arg: String): Option[ParsedWord] = arg.toList match {
+  def parseWord(arg: String): List[ParsedWord] = arg.toList match {
     case '-' :: '-' :: w => w.indexOf('=') match {
-      case -1 => Some(ParsedWord(LongParsedName(w.mkString), None))
+      case -1 => ParsedWord(LongParsedName(w.mkString), None).pure[List]
       case i  =>
         val (cmd, value) = w.splitAt(i)
-        Some(ParsedWord(LongParsedName(cmd.mkString), Some(value.tail.mkString)))
+        ParsedWord(LongParsedName(cmd.mkString), Some(value.tail.mkString)).pure[List]
     }
-    case '-' :: w :: Nil =>
-      Some(ParsedWord(ShortParsedName(w), None))
-    case _ =>
-      None
+    case '-' :: ws => ws map (w => ParsedWord(ShortParsedName(w), None))
+    case _ => Nil
   }
 
   def stepParser[A](s: ParseState, arg: String, p: Parse[A]): NondetT[StateArg, Parse[A]] =
     s match {
       case SkipOpts => parseWord(arg) match {
-        case None => searchArg(arg, p)
-        case Some(w) => searchOpt(w, p)
+        case Nil     => searchArg(arg, p)
+        case ws      => ws.foldLeft(NondetT.lift(p.pure[StateArg]))((a,b) => a flatMap( p1 => (searchOpt(b, p1) )))
```

Comment:
Both this and the one below can be [foldLeftM](https://github.com/scalaz/scalaz/blob/466d4da2951666f736d554104a2b3b956e4e6885/core/src/main/scala/scalaz/Foldable.scala#L64) /fyi /useful